main.go

```diff
@@ -0,0 +1,419 @@
+		Mob:  r.FormValue("contact_numbers"),
+	}
+	if err:= json.NewEncoder(w).Encode(b); err != nil {
+		fmt.Println("ERRORRRRR : " + err.Error())
+		http.Error(w,err.Error(),http.StatusInternalServerError)
+	}
+
+}
+
+func (obj OneContact) ServeOneContact(w http.ResponseWriter, r *http.Request){
+	P := []OneContact{}
+	fmt.Println("One Contact: ",gmux.Vars(r)["name"])
+	rows,err:= db.Query("select * from numbers where name=?",gmux.Vars(r)["name"])
+	if err!=nil {
+		http.Error(w,err.Error(),http.StatusInternalServerError)
+
+	}
+	for rows.Next(){
+		err = rows.Scan(&obj.Name,&obj.Mob)
+		if err!=nil {
+			http.Error(w,err.Error(),http.StatusInternalServerError)
+		}
+		 P = append(P,obj)
+	}
+
+	fmt.Println("MOBILEEEE : ",P)
+	//staticPages.Execute(w,P)
+	/*b:= Contacts{
+		Name: gmux.Vars(r)["name"],
+		Mob:  obj.Mob,
+	}*/
+	if err:= json.NewEncoder(w).Encode(P); err != nil {
+		fmt.Println("ERRORRRRR One contact: " + err.Error())
+		http.Error(w,err.Error(),http.StatusInternalServerError)
+	}
+}
+
+
+func (obj Contacts) serveAdd(w http.ResponseWriter, r *http.Request)  {
+	fmt.Println("Did you come here ?")
+	result, err := db.Exec("INSERT INTO contacts (name,mob,email,address,nationality,user_Name) VALUES (?, ?, ?,?,?,?) ",
+		r.FormValue("name"), r.FormValue("mob"), r.FormValue("email"),r.FormValue("address"), r.FormValue("nationality"),sessions.GetSession(r).Get("username"))
+	_,err1 := db.Exec("insert into numbers (name,contact_numbers) values (?,?) ",r.FormValue("name"),r.FormValue("mob"))
+	if err != nil || err1 !=nil {
+		fmt.Println("ERROR Contacts/add :" + err.Error())
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+
+	fmt.Println("RESULT : ", result)
+	x,_ := result.LastInsertId()
+	fmt.Println("LAST INSERTED ROW: ",x)
+	b:= Contacts{
+		Name: r.FormValue("name"),
+		Mob:  r.FormValue("mob"),
+		Email:r.FormValue("email"),
+		Address:r.FormValue("address"),
+		Nationality:r.FormValue("nationality"),
+	}
+	if err:= json.NewEncoder(w).Encode(b); err != nil {
+		fmt.Println("ERRORRRRR : " + err.Error())
+		http.Error(w,err.Error(),http.StatusInternalServerError)
+	}
+}
+
+func (obj User) serveLogin(w http.ResponseWriter, r *http.Request)  {
+	fmt.Println("Did you SERVE LOGIN ?")
+	var p LoginPage
+	if r.FormValue("register") != ""{
+		fmt.Println("hello From Register: ",r.FormValue("register"))
+		//if we need to register we'll need a bcrypt hash from the given pass
+		secret,_ := bcrypt.GenerateFromPassword([]byte(r.FormValue("password")),bcrypt.DefaultCost)
+		s := string(secret[:])
+		fmt.Println("SECRET PASS = ",s)
+		user := User{r.FormValue("username"),secret} //new user object
+
+		fmt.Println("hello from Train Register two :",r.FormValue("username"))
+		fmt.Println("Let's Print user : ",user.UserName)
+		fmt.Println("Let's print Pass : ",string(user.Secret))
+
+
+		_, err := db.Exec("INSERT INTO users (username,secret) VALUES (?, ?)", user.UserName, user.Secret)
+		if err != nil {
```

Comment:
on Db i made it a unique user so when adding user with same name before it'll report an error